src/extension.ts

```diff
@@ -383,22 +367,70 @@ async function terraformCommand(command: string, languageServerExec = true): Pro
 }
 
 function enabled(): boolean {
-  return config('terraform').get('languageServer.external', false);
+  return config('terraform').get('languageServer.enable', false);
 }
 
-async function migrateLegacySettings() {
-  if (config('terraform').has('languageServer.enabled')) {
-    try {
-      await config('terraform').update(
-        'languageServer',
-        { enabled: undefined, external: true },
-        vscode.ConfigurationTarget.Global,
-      );
-    } catch (err) {
-      const error = err instanceof Error ? err.message : err;
-      console.error(`Error trying to erase pre-2.0.0 settings: ${error}`);
-    }
+async function migrateLegacySettings(ctx: vscode.ExtensionContext) {
+  ctx.globalState.update('terraform.disableSettingsMigration', false);
+  if (ctx.globalState.get('terraform.disableSettingsMigration', false)) {
+    return;
+  }
+
+  const warnMigration = await warnIfMigrate([
+    'terraform.languageServer.external',
+    'terraform.languageServer.pathToBinary',
+    'terraform-ls.rootModules',
+    'terraform-ls.excludeRootModules',
+    'terraform-ls.ignoreDirectoryNames',
+    'terraform-ls.terraformExecPath',
+    'terraform-ls.terraformExecTimeout',
+    'terraform-ls.terraformLogFilePath',
+    'terraform-ls.experimentalFeatures',
+  ]);
+  if (warnMigration === false) {
+    return;
   }
+
+  const choice = await vscode.window.showInformationMessage(
+    'Terraform configuration settings have changed in the latest update',
+    {
+      detail: 'You can automatically migrate or ask to not be reminded again and migrate the settings yourself',
+      modal: true,
+    },
+    { title: 'Migrate' },
+    { title: "Don't remind me again" },
+  );
+  if (choice === undefined) {
+    return;
+  }
+
+  switch (choice.title) {
+    case "Don't remind me again":
+      ctx.globalState.update('terraform.disableSettingsMigration', true);
+      return;
+    case 'Migrate':
+    // migrate below
+  }
+
```

Comment:
The migration UI looks neat, but it didn't seem to do anything for me once confirmed

<img width="253" alt="Screenshot 2022-06-23 at 12 33 42" src="https://user-images.githubusercontent.com/287584/175280459-7bc71a19-54d0-4ba6-aee7-d3dac7cafa06.png">

<img width="677" alt="Screenshot 2022-06-23 at 14 59 00" src="https://user-images.githubusercontent.com/287584/175304547-796cf6ab-d798-4c76-aab6-d6fb0011dd40.png">

Not sure if that has to do anything with the dev mode - do I need to package a vsix and install it properly for this to work?

Reply:
No, you should be able to do this in the extension host in dev mode. Was that example in user, folder or workspace settings?

Reply:
This is user, i.e. JSON file in `~/Library/Application Support/Code/User/settings.json`
I'm happy to jump on Zoom to look into this further - it may be something on my side.